feat(filters): implement Search, Category & Occasion filters with NgRx

SaadGomaa opened the pull request:
- Adds product filtering with NgRx — keyword search, multi-category & occasion toggles, and a “Clear All” reset.
- All logic lives in a new filters slice.
- selectFilteredProducts combines products × filters for instant UI updates without extra API calls.

Commit:
feat(filters): implement Search, Category & Occasion filters with NgRx

aashraf2510 commented on apps/flower-app/src/app/store/composite-selectors/filtered-products.selectors.ts:
```diff
@@ -0,0 +1,15 @@
+  selectAllProducts,
+  selectFilters,
+  (products, f) =>
+    products.filter((p) => {
```
Filter process should be handled in reducers not in selectors  

Add filtered products in the products store , and get filtered products using selector

SaadGomaa commented:
@aashraf2510 Done:
- Moved all filtering logic into **products.reducer.ts** (`calcFilteredIds` + `Apply Filters` action).  
- Added `filteredIds` to the products state; selector `selectFilteredProducts` now just maps those IDs.  
- Removed the old `filtered-products.selectors.ts` file (no filtering in selectors anymore).

Let me know if anything else needs tweaking

Commit:
refactor(filters): move filtering to reducer + add filteredIds

SaadGomaa commented:
@aashraf2510  changes pushed 
– please take another look

SaadGomaa committed:
Merge branch 'main' into feat/filtersWithNGRX

Commit:
resolve conflicts with main

Commit:
Merge branch 'feat/filtersWithNGRX' of https://github.com/maomenraafat/rose-ecommerce-apps into feat/filtersWithNGRX

SaadGomaa commented:
@aashraf2510 Conflicts resolved
— please approve when you can
